### ISSUE TITLE
Issue with setup_tax function in EDD_Payment class

### ISSUE THREAD

**Issue by bseddon** [easydigitaldownloads/easy-digital-downloads#5100]
Line 1843 of class [EDD_Payment](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/master/includes/payments/class-edd-payment.php) is incorrect.  It's trying to access a local variable that is not declared.  No error is presented because the use of this variable is wrapped in a call to isset().

`$tax = isset( $payment_meta['tax'] ) ? $payment_meta['tax'] : 0;`

The call should be:

`$tax = isset( $this->payment_meta['tax'] ) ? $this->payment_meta['tax'] : 0;`

I'm sure this issue will not affect most users because the earlier call to retrieve the tax value from the meta value '_edd_payment_tax' will work.  I noticed the problem with line 1843 because I'm currently working with a database which have discovered an earlier update process failed to run correctly.

**Comment by cklosowski** [comment]
@bseddon does this fix it up for you? Is it possible you could provide us the code you're trying to run where you come about this?

**Comment by bseddon** [comment]
Yes, that works.  

I don't have any example code to provide you.  Last week I was messing around with some code for our VAT plugin where I had a need to access payment information across a range of payments. Using the EDD_Payment class seemed to be the way to go.  In using it to access the tax balance from a selection of historic payments I saw some of the payments were showing a zero balance.  I knew this was not correct.  

Usning the function edd_get_payment_meta() to look at the cart metadata I was able to confirm the tax was really there.  That prompted me to find out why the discrepancy exists.  The answer was that the affected payments did not have _edd_payment_tax and _edd_payment_total meta records.  We think because we managed to skip an update process.

Because the meta records did not exist, the condition in the setup_tax() method was triggered but, of course, it didn't work to retrieve the tax the 'old' way.
